geonode/qgis_server/views.py

```diff
@@ -173,6 +173,10 @@ def download_map(request, mapid):
         if map_layer.local:
             layer = get_object_or_404(Layer, alternate=map_layer.name)
             qgis_layer = get_object_or_404(QGISServerLayer, layer=layer)
+
+            # try getting qml from qgis-server
+            qgis_layer.extract_default_style_to_qml()
```

Comment:
use:

```
with qgis_layer.use_default_style_as_qml():
    # the entire bulk of code that uses qgis_layer.qml_path

# when it exited `with` statements it will automatically deleted qml
# you don't have to call qgis_layer.remove_qml_file_style()

Reply:
this is analoguos with how you open file in pythonic way with `with` statement

```
with open('/some/files.xml') as f:
    # do something with the files

# when it exits, the file is closed automatically
```

Reply:
thanks @lucernae 
i've changed it accordingly.